src/celeritas/ext/detail/AllElementReader.hh

```diff
@@ -42,6 +42,7 @@ class AllElementReader
 
     //! Load a map of data for all stored elements
     template<class ReadOneElement>
+    // NOLINTNEXTLINE(cppcoreguidelines-missing-std-forward)
```

Comment:
This thing is obnoxious.  LLVM algorithm implementations don't follow it:
https://github.com/llvm/llvm-project/blob/bb9bb686741feeb2c65097ef750126477f96ff30/libcxx/include/__algorithm/make_heap.h#L31

but should we instead omit the `&&`? (Would that copy objects rather than referencing them?)
https://github.com/llvm/llvm-project/blob/bb9bb686741feeb2c65097ef750126477f96ff30/libcxx/include/__algorithm/for_each_n.h#L26

Reply:
Yeah it would make a copy, though in the typical use case, you pass in a temporary. So, the universal reference is actually deduced to a value type, and directly constructs the argument in place. I guess we can keep it as is for uniformity